File: redwood-yoga-java/src/main/java/io/github/orioncraftmc/meditate/YogaMeasureOutput.java
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package io.github.orioncraftmc.meditate;

import io.github.orioncraftmc.meditate.internal.YGSize;

/**
 * Helpers for building measure output value.
 */
public class YogaMeasureOutput {

  public static YGSize make(float width, float height) {
    return new YGSize(width, height);
  }

  public static YGSize make(int width, int height) {
    return make((float) width, (float) height);
  }

  public static float getWidth(YGSize measureOutput) {
    return measureOutput.width;
  }

  public static float getHeight(YGSize measureOutput) {
    return measureOutput.height;
  }
}

File: redwood-yoga/src/commonMain/kotlin/app/cash/redwood/yoga/YogaMeasureOutput.kt
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
package app.cash.redwood.yoga

import app.cash.redwood.yoga.internal.YGSize
import kotlin.jvm.JvmStatic

/**
 * Helpers for building measure output value.
 */
object YogaMeasureOutput {
  @JvmStatic
  fun make(width: Float, height: Float): YGSize {
    return YGSize(width, height)
  }

  @JvmStatic
  fun make(width: Int, height: Int): YGSize {
    return make(width.toFloat(), height.toFloat())
  }

  @JvmStatic
  fun getWidth(measureOutput: YGSize): Float {
    return measureOutput.width
  }

  @JvmStatic
  fun getHeight(measureOutput: YGSize): Float {
    return measureOutput.height
  }
}
